### PR TITLE
feat: add multiple additive scenes feature

### PR DESCRIPTION
**Pull request by Moddingdudes**
This will internally calculate if the server should load another additive scene, or only load once.

### CODE REVIEW

File: Assets/Mirage/Runtime/NetworkSceneManager.cs [hunk 1]
```diff
@@ -32,6 +32,10 @@ public class NetworkSceneManager : MonoBehaviour
         [FormerlySerializedAs("server")]
         public NetworkServer Server;
 
+        [FormerlySerializedAs("loadMultipleAdditive")]
+        [Tooltip("If true, the server will create a new additive scene for every client that joins, and that client will receive all loaded additive scenes upon joining")]
+        public bool loadMultipleAdditiveScenes = true;
```
Comment (James-Frowen) [inline Assets/Mirage/Runtime/NetworkSceneManager.cs:37]: Can you rename this to `LoadAdditiveScenes`

File: Assets/Mirage/Runtime/NetworkSceneManager.cs [hunk 2]
```diff
@@ -358,18 +362,30 @@ private UniTask ServerSceneLoading(string scenePath, IEnumerable<INetworkPlayer>
...
             // Coburn, 2023-01-29: Patched the following to not use Forget UniTask functionality.
+            // Moddingdudes, 2023-02-06: This is only for multiple additive, we want just one additive option
             // ALWAYS load the scene, even if we're the host.
             // return the task after sending the message
-            var loadTask = LoadSceneAsync(scenePath, players, sceneOperation, loadSceneParameters);
+
+            UniTask loadTask = UniTask.CompletedTask;
+
+            if (serverShouldLoad)
+                loadTask = LoadSceneAsync(scenePath, players, sceneOperation, loadSceneParameters);
```
Comment (James-Frowen) [inline Assets/Mirage/Runtime/NetworkSceneManager.cs:388]: Looks like loadMultipleAdditiveScenes is doing something different here than in OnServerAuthenticated. 

If it is to load scene on client but not server then it should probably be a new function

File: Assets/Mirage/Runtime/NetworkSceneManager.cs [hunk 3]
```diff
@@ -358,18 +362,30 @@ private UniTask ServerSceneLoading(string scenePath, IEnumerable<INetworkPlayer>
...
 
-            OnServerStartedSceneChange?.Invoke(scenePath, sceneOperation);
 
-            if (Server.LocalClientActive)
+            bool serverSceneIsLoaded = GetSceneByPathOrName(scenePath).isLoaded;
+
+            //Only do a guarantee load if we are loading multiple additive scenes, otherwise only load if unloaded
+            bool serverShouldLoad = loadMultipleAdditiveScenes ? true : !serverSceneIsLoaded;
+
+            if (serverShouldLoad)
```
Comment (James-Frowen) [inline Assets/Mirage/Runtime/NetworkSceneManager.cs:371]: 3 of these `if` blocks check `if (serverShouldLoad)` they should probably just be 1 block instead

File: Assets/Mirage/Runtime/NetworkSceneManager.cs [hunk 4]
```diff
@@ -32,6 +32,10 @@ public class NetworkSceneManager : MonoBehaviour
         [FormerlySerializedAs("server")]
         public NetworkServer Server;
 
+        [FormerlySerializedAs("loadMultipleAdditive")]
+        [Tooltip("If true, the server will create a new additive scene for every client that joins, and that client will receive all loaded additive scenes upon joining")]
```
[the remark this answers is not in the document]
Comment (Moddingdudes) [reply Assets/Mirage/Runtime/NetworkSceneManager.cs:36]: Can change it to that for now, but I feel like that only tells half the story. New users may think that it only loads the additive scenes for them, and doesn't load a new one on the server every time and ideally isolates clients with scene interest management.

File: Assets/Mirage/Runtime/NetworkSceneManager.cs (hunk 2 above)
Comment (Moddingdudes) [reply Assets/Mirage/Runtime/NetworkSceneManager.cs:388]: You're right, adding another variable which separates the logic

**Comment by James-Frowen** [comment]
merged here https://github.com/MirageNet/Mirage/commit/f39414be4f072981dd75612ef526cd0eb8aa3f91

**Pull request closed without merging**